code/modules/mob/living/basic/vermin/butterfly.dm

```diff
@@ -44,3 +44,48 @@
 
 /mob/living/basic/butterfly/lavaland
 	unsuitable_atmos_damage = 0
+
+/mob/living/basic/butterfly/lavaland/temporary
+	name = "strange butterfly"
+	basic_mob_flags = DEL_ON_DEATH
+	/// The atom that's spawning the butterflies
+	var/atom/source = null
+	/// Max distance in tiles before the butterfly despawns
+	var/max_distance = 5
+	/// Whether the butterfly will be destroyed at the end of its despawn timer
+	var/will_be_destroyed = FALSE
+	/// Despawn timer of the butterfly
+	var/despawn_timer = 0
+
+/mob/living/basic/butterfly/lavaland/temporary/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+/mob/living/basic/butterfly/lavaland/temporary/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	. = ..()
+
+/mob/living/basic/butterfly/lavaland/temporary/process()
+	if(should_despawn())
+		if(will_be_destroyed)
+			return
+		will_be_destroyed = TRUE
+		despawn_timer = addtimer(CALLBACK(src, TYPE_PROC_REF(/mob/living/basic/butterfly/lavaland/temporary, fadeout)), 5 SECONDS, TIMER_STOPPABLE)
+	else
```

Comment:
You can early return ``should_despawn()`` instead of using an else, if you want. This proc doesn't do anything else so it would look nicer